homeassistant/components/minecraft_server/sensor.py

```diff
@@ -0,0 +1,265 @@
+    UNIT_VERSION,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, config_entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up the Minecraft Server sensor platform."""
+    server = hass.data[DOMAIN][config_entry.unique_id]
+
+    # Create entities list.
+    entities = [
+        MinecraftServerDescriptionSensor(hass, server),
+        MinecraftServerVersionSensor(hass, server),
+        MinecraftServerProtocolVersionSensor(hass, server),
+        MinecraftServerLatencyTimeSensor(hass, server),
+        MinecraftServerPlayersOnlineSensor(hass, server),
+        MinecraftServerPlayersMaxSensor(hass, server),
+        MinecraftServerPlayersListSensor(hass, server),
+    ]
+
+    # Add sensor entities.
+    async_add_entities(entities, True)
+
+
+class MinecraftServerSensorEntity(MinecraftServerEntity):
+    """Representation of a Minecraft Server sensor base entity."""
+
+    def __init__(
+        self,
+        hass: HomeAssistantType,
+        server: MinecraftServer,
+        type_name: str,
+        icon: str = None,
+        unit: str = None,
+        device_class: str = None,
+    ) -> None:
+        """Initialize sensor base entity."""
+        super().__init__(hass, server, type_name, icon, device_class)
+        self._state = None
+        self._unit = unit
+
+    @property
+    def state(self) -> Any:
+        """Return sensor state."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self) -> str:
+        """Return sensor measurement unit."""
+        return self._unit
+
+
+class MinecraftServerDescriptionSensor(MinecraftServerSensorEntity):
+    """Representation of a Minecraft Server description sensor."""
+
+    _COLOR_CODES = [
+        "§0",
+        "§1",
+        "§2",
+        "§3",
+        "§4",
+        "§5",
+        "§6",
+        "§7",
+        "§8",
+        "§9",
+        "§a",
+        "§b",
+        "§c",
+        "§d",
+        "§e",
+        "§f",
+        "§k",
+        "§l",
+        "§m",
+        "§n",
+        "§o",
+        "§r",
+    ]
+
+    def __init__(self, hass: HomeAssistantType, server: MinecraftServer) -> None:
+        """Initialize description sensor."""
+        super().__init__(
+            hass=hass,
+            server=server,
+            type_name=NAME_DESCRIPTION,
+            icon=ICON_DESCRIPTION,
+            unit=UNIT_DESCRIPTION,
+        )
+
+    async def async_update(self) -> None:
+        """Update description."""
+        description = self._server.description
+
+        if description is not None:
+            # Remove color codes.
+            for color_code in self._COLOR_CODES:
+                description = description.replace(color_code, "")
+
+            # Remove newlines.
+            description = description.replace("\n", " ")
+
+            # Limit description length to 255.
+            if len(description) > 255:
+                description = description[:255]
```

Comment:
Do we really need this sensor? How will it be used?

Reply:
@Quentame also asked this 😄 One use case I was thinking of is for notifications, like: 

> Someone joined the server DESCRIPTION.

I thought this is easier with a sensor. One solution would be for sure to have it as a state attribute in another sensor, but this would require more templating skills of the user, or not? Looking forward for your opinion.

Reply:
If we want notifications I suggest adding a notify platform, **in another PR** later.

Reply:
For now I suggest we don't add this sensor.

Reply:
Maybe just fire custom events with the description as event data. Then user's can trigger automations as wanted on the event data.

Reply:
So yeah, again, for now, I think we should not add this sensor.